### PR TITLE
Only return unarchived bedspace count from getPremisessSummary

### PR DESCRIPTION
**Pull request by marcocolluramoj**
Adds a condition to only include in the bedspace count returned from `premisesSummaryGet` the number of bedspaces with no end date or an end date in the future.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt [hunk 1]
```diff
@@ -67,6 +67,112 @@ class PremisesSummaryTest : IntegrationTestBase() {
     }
   }
 
+  @Test
+  fun `Get all CAS3 Premises returns a bedspace count of 3 when there are 3 bedspaces with no end date and one 1 with an end date in the past`() {
```
Comment (muhammad-elabdulla) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt:71]: I will suggest to change the method name to:
Get all CAS3 Premises returns bedspace count as expected when there is an archived bedspace

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt [hunk 2]
```diff
@@ -67,6 +67,112 @@ class PremisesSummaryTest : IntegrationTestBase() {
...
+        .jsonPath("$[0].postcode").isEqualTo("NW1 6XE")
+        .jsonPath("$[0].status").isEqualTo("active")
+        .jsonPath("$[0].pdu").isEqualTo(expectedCas3Premises.probationDeliveryUnit!!.name)
+        .jsonPath("$[0].bedCount").isEqualTo(3)
+        .jsonPath("$.length()").isEqualTo(1)
+    }
+  }
+
+  @Test
+  fun `Get all CAS3 Premises returns a bedspace count of 4 when there are 3 bedspaces with no end date and one 1 with an end date in the future`() {
```
Comment (muhammad-elabdulla) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt:124]: I suggest to change the test name to:
Get all CAS3 Premises returns a bedspace count as expected when beds are active

**Review by muhammad-elabdulla — APPROVED** [review]

**Comment by muhammad-elabdulla** [comment]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt (hunk 2 above)
Comment (marcocolluramoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesSummaryTest.kt]: done

**Pull request merged**